crates/router/src/core/payments/operations/payments_device_data_collection.rs

```diff
@@ -0,0 +1,310 @@
+use std::marker::PhantomData;
+
+use api_models::enums::FrmSuggestion;
+use async_trait::async_trait;
+use error_stack::ResultExt;
+use router_derive::PaymentOperation;
+use router_env::{instrument, tracing};
+
+use super::{BoxedOperation, Domain, GetTracker, Operation, UpdateTracker, ValidateRequest};
+use crate::{
+    core::{
+        errors::{self, CustomResult, RouterResult, StorageErrorExt},
+        payments::{
+            helpers, operations, CustomerDetails, PaymentAddress, PaymentData,
+            PaymentMethodRetrieve,
+        },
+    },
+    db::StorageInterface,
+    routes::AppState,
+    services,
+    types::{
+        api::{self, PaymentIdTypeExt},
+        domain,
+        storage::{self, enums as storage_enums},
+    },
+    utils::OptionExt,
+};
+
+#[derive(Debug, Clone, Copy, PaymentOperation)]
+#[operation(ops = "all", flow = "ddc")]
+pub struct PaymentDeviceDataCollection;
+
+#[async_trait]
+impl<F: Send + Clone, Ctx: PaymentMethodRetrieve>
+    GetTracker<F, PaymentData<F>, api_models::payments::PaymentsDeviceDataCollectionRequest, Ctx>
+    for PaymentDeviceDataCollection
+{
+    #[instrument(skip_all)]
+    async fn get_trackers<'a>(
+        &'a self,
+        state: &'a AppState,
+        payment_id: &api::PaymentIdType,
+        _request: &api_models::payments::PaymentsDeviceDataCollectionRequest,
+        _mandate_type: Option<api::MandateTransactionType>,
+        merchant_account: &domain::MerchantAccount,
+        mechant_key_store: &domain::MerchantKeyStore,
+        _auth_flow: services::AuthFlow,
+    ) -> RouterResult<(
+        BoxedOperation<'a, F, api_models::payments::PaymentsDeviceDataCollectionRequest, Ctx>,
+        PaymentData<F>,
+        Option<CustomerDetails>,
+    )> {
+        let (mut payment_intent, payment_attempt, currency, amount);
+        let db = &*state.store;
+
+        let merchant_id = &merchant_account.merchant_id;
```

Comment:
can you verify if you need to build all the entities here as we are going to build only the HTML response for this flow